packages/utils/src/cmds/transform_translate.js

```diff
@@ -0,0 +1,381 @@
+            'Create the output dir if not existing, use `--no-create-out-dir` to exit with 1 when folder does not exist',
+        type: 'boolean',
+        default: 'true',
+    },
+
+    overrideExistingFiles: {
+        describe: 'Overriding the contents of existing translation files',
+        type: 'boolean',
+        default: 'false',
+    },
+
+    appendToExistingFiles: {
+        describe:
+            'Appends the new contents to existing translation files, can only be used when not using `--override-existing-files`',
+        type: 'boolean',
+        default: 'false',
+    },
+
+    logMissingKeys: {
+        describe:
+            'Log keys that are present in translation files which are not present in translation file of --primary-language',
+        type: 'boolean',
+        default: 'false',
+    },
+
+    deleteOldFiles: {
+        describe:
+            'Delete the old files that were transformed (will only delete files specified with the `--language` option when present)',
+        type: 'boolean',
+    },
+}
+exports.handler = argv => {
+    const inDirExists = fs.existsSync(argv.inDir)
+    const outDirExists = fs.existsSync(argv.outDir)
+    const languagesToTransform = argv.languages
+        ? argv.languages.split(/,\s*/)
+        : []
+
+    if (!argv.appName) {
+        log.error(`You need to provied an app name`)
+        process.exit(1)
+    }
+
+    if (!appNamePattern.test(argv.appName)) {
+        log.error(exports.builder.appName.describe)
+        log.error(`Received: ${argv.appName}`)
+        process.exit(1)
+    }
+
+    if (!inDirExists) {
+        log.error(`Input directory does not exist ("${argv.inDir}")`)
+        process.exit(1)
+    } else {
+        argv.verbose && log.debug(`Input directory exists ("${argv.inDir}")`)
+    }
+
+    if (argv.noCreateOutDir && !outDirExists) {
+        log.error(`Output directory does not exist ("${argv.outDir}")`)
+        process.exit(1)
+    } else if (!argv.noCreateOutDir && !outDirExists) {
+        argv.verbose && log.debug(`Creating output dir: ${argv.outDir}`)
+        fs.mkdirSync(argv.outDir, { recursive: true })
+    } else {
+        argv.verbose && log.debug('Output dir already exists, skip creating it')
+    }
+
+    const translationFiles = fs
+        .readdirSync(argv.inDir)
+        .filter(fileIsOldTranslationFile)
+
+    const mainTranslationFile = translationFiles.find(file =>
+        file.match(`i18n_module_${argv.primaryLanguage}.properties`)
+    )
+    const mainTranslationFilePath = path.join(argv.inDir, mainTranslationFile)
+
+    if (!fs.existsSync(mainTranslationFilePath)) {
+        log.error(
+            `Main language file must exist ("${path.join(
+                argv.inDir,
+                mainTranslationFile
+            )}")`
+        )
+        process.exit(1)
+    }
+
+    argv.verbose &&
+        log.debug('Extracting key/value pairs from translation files')
+    const translations = translationFiles.reduce((mappings, file) => {
```

Comment:
It would be nice to factor this function a bit, could we move everything that's not argument parsing logic out into a separate file and separate out, for instance, the body of this reduce function so everything isn't inline here?